process/block/interceptedBlocks/interceptedEquivalentProof.go

```diff
@@ -136,20 +137,7 @@ func (iep *interceptedEquivalentProof) checkHeaderParamsFromProof() error {
 		return fmt.Errorf("%w while getting header for proof hash %s", err, hex.EncodeToString(iep.proof.GetHeaderHash()))
 	}
 
-	if iep.proof.GetHeaderNonce() != header.GetNonce() {
-		return fmt.Errorf("%w, nonce mismatch", ErrInvalidProof)
-	}
-	if iep.proof.GetHeaderShardId() != header.GetShardID() {
-		return fmt.Errorf("%w, shard id mismatch", ErrInvalidProof)
-	}
-	if iep.proof.GetHeaderEpoch() != header.GetEpoch() {
-		return fmt.Errorf("%w, epoch mismatch", ErrInvalidProof)
-	}
-	if iep.proof.GetHeaderRound() != header.GetRound() {
-		return fmt.Errorf("%w, round mismatch", ErrInvalidProof)
-	}
-
-	return nil
+	return common.VerifyProofAgainstHeader(iep.proof, header)
```

Comment:
I think we need to add a method (it there is none already) to get the header by hash from pool and if not found from storer (cache and if not found from storage)

Reply:
pushed